### ISSUE TITLE
PROJ: Automatically Generate a Contributors list in MAINTAINERS.md file

### ISSUE THREAD

**Issue by decause-gov** [DSACMS/open#4]
## Mentor
@decause-gov @IsaacMilarky 

## Brief Description

See Below

## Scoping
2 weeks

## Required Deliverables
 - GitHub Actions workflow script that generates a contributors list in `MAINTAINERS.md`file based on the contributors within a GitHub repository and commits it back to the repository.
 - Include a count of contributors too

## Desired Date Window: (June, July, August, Any)
June

## INSPO
https://github.com/marketplace/actions/contribute-list

**Comment by Gaelan** [comment]
It'll be difficult to distinguish "maintainers" from drive-by contributor or former maintainers, but a list of everyone who's committed to the repo wouldn't be too hard.

Existing tools that might be usable, or at least useful inspiration: https://allcontributors.org, https://github.com/marketplace/actions/contribute-list, https://github.com/marketplace/actions/contributor_list

**Comment by natalialuzuriaga** [comment]
https://github.com/marketplace/actions/contribute-list

**Comment by decause-gov** [comment]
06/21:
- This ticket depends on #97 in repo-scaffolder

**Comment by IsaacMilarky** [comment]
We can also talk to the RedHat CHAOSS people about how they determine if a contributor is 'active' or not. Since I know that CHAOSS has pretty good metrics when it comes to that info because its really useful

**Comment by Firebird1029** [comment]
I can take a look into that if you want me too — for now the PR just has everyone who has contributed at some point, but I can make a separate PR later on adding an active contributors section

Dropping PR link here: https://github.com/DSACMS/repo-scaffolder/pull/102

**Comment by decause-gov** [comment]
06/24:
- Switching from Rest API to GraphQL API
- goals: get this working in GH Actions (formatting issues)
- IDEA: export to a .gql file, and cat into a ENV variable?
- stretch: add this to one or more repos to test in prod!

**Comment by Firebird1029** [comment]
@IsaacMilarky @natalialuzuriaga ready for re-review: https://github.com/DSACMS/repo-scaffolder/pull/102

**Comment by IsaacMilarky** [comment]
6/28:

- Deliverables are here and they work!

**Comment by Firebird1029** [comment]
Tested on metrics repo here: https://github.com/DSACMS/metrics/blob/main/MAINTAINERS.md and it works

**Comment by decause-gov** [comment]
09/06:
- Murt put some cycles into this, will continue next week between travel.

**Comment by IsaacMilarky** [comment]
9/30:

- @IsaacMilarky will fix this issue this week along with the other tickets that need to be done on metrics this week
- Shouldn't be too major of a task to complete

**Comment by IsaacMilarky** [comment]
10/4:

- ACTION: @IsaacMilarky Will fix issues with this workflow today.

**Comment by natalialuzuriaga** [comment]
10/07:
- @IsaacMilarky is working on this!

**Comment by decause-gov** [comment]
10/21:
- Sachin is looking into this
- ACTION: Schedule Hacksesh w/Murt for context gathering

**Comment by IsaacMilarky** [comment]
10/22:
1. Deal with the branch protection issue with akhilmhdh/contributors-readme-action
2. Clean up the non akhilmhdh/contributors-readme-action part of the contributors.yml file (probably put it in an outside script file)
3.Deal with the branch protection for the last part

**Comment by decause-gov** [comment]
10/25:
- @sachin-panayil Forked a copy to do action testing
- ACTION: continue hacking on this next week!

**Comment by decause-gov** [comment]
11/01:
- discussed in 1x1 with decause
- blocked by GH-actions. Hacky solution for now, will connect async and/or next week.

**Comment by sachin-panayil** [comment]
11/12:
- Merged! https://github.com/DSACMS/repo-scaffolder/pull/204
